src/devices/Mpu9250/Mpu9250.cs

```diff
@@ -0,0 +1,983 @@
+                Vector3 gyro_temp = new Vector3();
+
+                // Read data
+                ReadByteArray(Register.FIFO_R_W, rawData);
+
+                // Form signed 16-bit integer for each sample in FIFO
+                accel_temp.X = BinaryPrimitives.ReadUInt16BigEndian(rawData);
+                accel_temp.Y = BinaryPrimitives.ReadUInt16BigEndian(rawData.Slice(2));
+                accel_temp.Z = BinaryPrimitives.ReadUInt16BigEndian(rawData.Slice(4));
+                gyro_temp.X = BinaryPrimitives.ReadUInt16BigEndian(rawData.Slice(6));
+                gyro_temp.Y = BinaryPrimitives.ReadUInt16BigEndian(rawData.Slice(8));
+                gyro_temp.Z = BinaryPrimitives.ReadUInt16BigEndian(rawData.Slice(10));
+
+                acceBias += accel_temp;
+                gyroBias += gyro_temp;
+            }
+            // Make the average
+            acceBias /= packetCount;
+            gyroBias /= packetCount;
+
+            // bias on Z is cumulative
+            acceBias.Z += acceBias.Z > 0 ? -AccSensitivity : AccSensitivity;
+
+            // Divide by 4 to get 32.9 LSB per deg/s 
+            // Biases are additive, so change sign on calculated average gyro biases
+            rawData[0] = (byte)(((int)(-gyroBias.X / 4) >> 8) & 0xFF);
+            rawData[1] = (byte)((int)(-gyroBias.X / 4) & 0xFF);
+            rawData[2] = (byte)(((int)(-gyroBias.Y / 4) >> 8) & 0xFF);
+            rawData[3] = (byte)((int)(-gyroBias.Y / 4) & 0xFF);
+            rawData[4] = (byte)(((int)(-gyroBias.Z / 4) >> 8) & 0xFF);
+            rawData[5] = (byte)((int)(-gyroBias.Z / 4) & 0xFF);
+
+            // Changes all Gyroscope offsets
+            WriteRegister(Register.XG_OFFSET_H, rawData[0]);
+            WriteRegister(Register.XG_OFFSET_L, rawData[1]);
+            WriteRegister(Register.YG_OFFSET_H, rawData[2]);
+            WriteRegister(Register.YG_OFFSET_L, rawData[3]);
+            WriteRegister(Register.ZG_OFFSET_H, rawData[4]);
+            WriteRegister(Register.ZG_OFFSET_L, rawData[5]);
+
+            // Output scaled gyro biases for display in the main program
+            _gyroscopeBias = gyroBias / GyroSensitivity;
+
+            // Construct the accelerometer biases for push to the hardware accelerometer
+            // bias registers. These registers contain factory trim values which must be
+            // added to the calculated accelerometer biases; on boot up these registers
+            // will hold non-zero values. In addition, bit 0 of the lower byte must be
+            // preserved since it is used for temperature compensation calculations.
+            // Accelerometer bias registers expect bias input as 2048 LSB per g, so that
+            // the accelerometer biases calculated above must be divided by 8.
+
+            // A place to hold the factory accelerometer trim biases
+            Vector3 accel_bias_reg = new Vector3();
+            Span<byte> accData = stackalloc byte[2] { 0, 0 };
```

Comment:
NIT: I don't think you have to initialize the Span as it will get overriden in the next Read()